### PR TITLE
chore: emit PROXY_REPOSITORY_CREATED event when creating new repo

### PR DESCRIPTION
**Pull request by daveleek**
## About the changes

- Emits a new event on the eventBus when Proxy-service creates a new repository for a frontend token
- Adds a prometheus metrics counter for created proxy-repositories

image

### CODE REVIEW

**Review by FredrikOseberg — APPROVED** [review]
LGTM

File: src/lib/services/proxy-service.ts [hunk 1]
```diff
@@ -109,6 +110,7 @@ export class ProxyService {
         if (!client) {
             client = this.createClientForProxyToken(token);
             this.clients.set(token.secret, client);
+            this.config.eventBus.emit(PROXY_REPOSITORY_CREATED, {});
```
Comment (FredrikOseberg) [inline src/lib/services/proxy-service.ts:113]: Is the second argument needed?

File: src/lib/metrics.ts [hunk 2]
```diff
@@ -227,6 +227,10 @@ export default class MetricsMonitor {
             name: 'event_created_by_migration_count',
             help: 'Event createdBy migration count',
         });
+        const proxyRepositoriesCreated = createCounter({
+            name: 'proxy_repositories_created',
+            help: 'Proxy repositories created',
+        });
```
[the remark this answers is not in the document]
Comment (daveleek) [reply src/lib/metrics.ts:233]: I guess this one should be expected, but is also out of scope to address right now

File: src/lib/services/proxy-service.ts (hunk 1 above)
Comment (daveleek) [reply src/lib/services/proxy-service.ts:113]: I didn't have it there initially, but I couldn't find example of code that didn't provide something. And I wondered if that had anything to do with snapshot suddenly failing. But I'll remove it

**Pull request merged**